packages/core/src/lib/filter/index.ts

```diff
@@ -335,12 +125,13 @@ class Filter extends BaseProtocol implements IReceiver {
    * Instead, use `createSubscription` to create a new subscription.
    */
   async subscribe<T extends IDecodedMessage>(
```

Comment:
I think we can avoid breaking this API by adding following code:
```ts
for (let decoder of decodersArray) {
  subscription.addEventListener(decoder.contentTopic, callback);
}
```

Reply:
that is why the comments:
```
  /**
   * This method is used to satisfy the `IReceiver` interface.
   *
   * @hidden
   *
   * @param decoders The decoders to use for the subscription.
   * @param callback The callback function to use for the subscription.
   * @param opts Optional protocol options for the subscription.
   *
   * @returns A Promise that resolves to a function that unsubscribes from the subscription.
   *
   * @remarks
   * This method should not be used directly.
   * Instead, use `createSubscription` to create a new subscription.
   */
   ```